packages-ts/gauntlet-terra-contracts/src/commands/contracts/ocr2/proposeOffchainConfig.ts

```diff
@@ -138,11 +140,16 @@ const beforeExecute: BeforeExecute<CommandInput, ContractInput> = (context) => a
 }
 
 const makeContractInput = async (input: CommandInput): Promise<ContractInput> => {
+  if (!process.env.SECRET) {
+    throw new Error('SECRET is not set in env!')
+  }
```

Comment:
I think this should be part of the `validateInput` fn.

Reply:
We normally use `validateInput` for logic checks, missing properties are checked when making command and contract input. I'll move it up to command input so it's the first thing to check.

Reply:
Then something is off as `validateInput` sounds like a place to check for any input errors (missing or invalid).

But also, `process.env` so late in command construction is not the way to go, as the input source needs to be abstracted. When we change loading the secret from an HTTP endpoint (example) nothing in the command should change, it still gets a secret on the input. How it got there is not relevant.

In any case, don't spend too much time on it now as this warrants a bigger refactoring.